AEP/aep-0000y.md

```diff
@@ -0,0 +1,74 @@
+# AEP : ArmoniK's Domain Definition
+
+|                   |ArmoniK Enhancement Proposal|
+---:                |:---
+**AEP**             | 
+**Title**           | Architecture for ArmoniK Extesion development
+**Author**          | Italo Aguiar <<iaguiar@aneo.fr>>, Quentin Delamea <<qdelamea@aneo.fr>>
+**Status**          | Draft
+**Type**            | Standard
+**Creation Date**   | 2024-06-04
+
+# Abstract
+
+This AEP proposes a comprehensive definition of ArmoniK's domain. A domain is an area of interest or control within our system. Our goal is to clearly define the core domain entities to facilitate further development.
+
+# Motivation
+
+The purpose of this document is to establish a clear and consistent definition of ArmoniK's domain, enabling other documents and standards to effectively reuse these concepts.
+
+# Rationale
+
+To describe more complex features of ArmoniK accurately, it is crucial to have a shared vocabulary and avoid redundancy. While some definitions already exist in the ArmoniK glossary, this AEP seeks to foster community discussion and approval to ensure a unified understanding and usage of these terms.
+
+# Specification
+
+## Worker
+
+User-developed containerized software capable of performing one or several tasks depending on its implementation. A worker can simply take input data and perform calculations on it to return a result. A worker can also submit new tasks that will be self-performed or handled by different workers or other instances of itself.
```

Comment:
I think we need to be more specific:
- a worker can do both: return results and submit additionnal tasks;
- a worker embeds an implementation of the communication protocols with the scheduling agent (in fact a worker is an implementation inside a kind of shell);